include/openmc/bank.h

```diff
@@ -21,13 +16,10 @@ namespace openmc {
 namespace simulation {
 
 extern std::vector<Particle::Bank> source_bank;
-extern std::vector<Particle::Bank> fission_bank;
-extern std::vector<Particle::Bank> secondary_bank;
-#ifdef _OPENMP
-extern std::vector<Particle::Bank> master_fission_bank;
-#endif
 
-#pragma omp threadprivate(fission_bank, secondary_bank)
+extern Particle::Bank* fission_bank;
```

Comment:
That's a good question, I was on the fence about that one. Pros and cons:

Pros of using vector:
- More consistent with the rest of the code
- A little less prone to memory leaks due to having the destructor, and takes something off the to-do list for simulation cleanup.

Cons of using vector:
- For our use case, other developers might be tempted at some point to use vector functions (e.g., push_back()) on it, which would not be allowed. The result of using any of those functions would be undefined behavior which might be hard for a developer to track down the cause of. My thought was having it be just a pointer (might) make a developer more curious as to how it was allocated, and how it is used elsewhere, so therefore might be less error prone. Basically, since we don't want people to use vector functions on this, a good way to explicitly disallow that is to use a pointer rather than a vector.
- This is not a particularly important point, but the fission bank will need to be an allocated pointer on GPU, so doing it this way will minimize code delta. My thought is that if we are making big changes to the fission bank now, might as well bite the bullet and get that done.

Overall, I would probably lean towards just leaving it as an allocated pointer, but I can totally understand if the vector option sounds better. It will definitely work either way, and I can definitely see how the memory safety logistics would be more important for maintainability.

Reply:
Ah, right. For the omp atomic update, vector push_back won't work :thinking: Ok, I'm on board then. With respect to memory safety, perhaps the best thing to do would be to use `std::unique_ptr<Particle::Bank[]>` rather than a raw pointer. This would allow us to use `std::make_unique` for creation and then `.reset()` for deletion.

Reply:
Also, where `fission_bank` is declared, it would be good to put a comment explaining why we _can't_ use a vector.

Reply:
Ah, that makes sense, I'll go the unique_ptr route and add some comments explaining how it is used.